src/System.Text.Json/tests/Serialization/Null.WriteTests.cs

```diff
@@ -30,14 +49,57 @@ public static void EnableIgnoreNullValuesOnWrite()
             JsonSerializerOptions options = new JsonSerializerOptions();
             options.IgnoreNullValues = true;
 
-            var obj = new TestClassWithInitializedProperties();
-            obj.MyString = null;
-            obj.MyInt = null;
-            obj.MyIntArray = null;
-            obj.MyIntList = null;
+            var obj = new TestClassWithInitializedProperties
+            {
+                MyString = null,
+                MyInt = null,
+                MyIntArray = null,
+                MyIntList = null,
+                MyObjectList = new List<object> { null },
+                MyListList = new List<List<object>> { new List<object> { null } },
+                MyDictionaryList = new List<Dictionary<string, string>> { new Dictionary<string, string>() { ["key"] = null } },
+                MyStringDictionary = new Dictionary<string, string>() { ["key"] = null },
+                MyObjectDictionary = new Dictionary<string, object>() { ["key"] = null },
+                MyStringDictionaryDictionary = new Dictionary<string, Dictionary<string, string>>() { ["key"] = null },
+                MyListDictionary = new Dictionary<string, List<object>>() { ["key"] = null },
+                MyObjectDictionaryDictionary = new Dictionary<string, Dictionary<string, object>>() { ["key"] = null }
+            };
+
+            string expectedJson =
+                    @"{" +
+                        @"""MyObjectList"":[null]," +
+                        @"""MyListList"":[[null]]," +
+                        @"""MyDictionaryList"":[" +
+                            @"{" +
+                                @"""key"":null" +
+                            @"}" +
+                        @"]," +
+                        @"""MyStringDictionary"":{" +
+                            @"""key"":null" +
+                        @"}," +
+                        @"""MyObjectDictionary"":{" +
+                            @"""key"":null" +
+                        @"}," +
+                        @"""MyStringDictionaryDictionary"":{" +
+                            @"""key"":null" +
+                        @"}," +
+                        @"""MyListDictionary"":{" +
+                            @"""key"":null" +
+                        @"}," +
+                        @"""MyObjectDictionaryDictionary"":{" +
+                            @"""key"":null" +
+                        @"}" +
+                    @"}";
 
-            string json = JsonSerializer.Serialize(obj, options);
-            Assert.Equal(@"{}", json);
+            Assert.Equal(expectedJson, JsonSerializer.Serialize(obj, options));
```

Comment:
Since reflection is used to determine property order during serialization, I believe this could fail in certain cases especially non-Windows. In other places we validate individual elements or just round-trip (object->json-object) so we can verify the object again.

Reply:
I've replaced with round-tripping.